test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -179,14 +179,14 @@ fs::path TransitionTestsSuite::suiteFillerFolder() const {
 	return fs::path("BlockchainTestsFiller") / "TransitionTests";
 }
 
-ChainBranch::ChainBranch(TestBlock const& _genesis): blockchain(_genesis)
+ChainBranch::ChainBranch(TestBlock const& _genesis, TestBlockChain::MiningType _miningType): blockchain(_genesis, _miningType)
 {
 	importedBlocks.push_back(_genesis);
 }
 
-void ChainBranch::reset()
+void ChainBranch::reset(TestBlockChain::MiningType _miningType)
 {
-	blockchain.reset(importedBlocks.at(0));
+    blockchain.reset(importedBlocks.at(0), _miningType);
```

Comment:
Is this ok that this `importedBlocks.at(0)` can stay the same even with the change of `_miningType`? i.e. Can the same `TestBlock` work for both types?

Reply:
possibly could make tests where valid block has invalid PoW and invalid block has valid PoW proof

Reply:
I meant here that you have one instance of `TestBlock` (namely `importedBlocks.at(0)`) which you can use after call to `reset` with different seal engine. That looked suspicious.

Reply:
yes. that is if you change _miningType inside the for loop. should set it to const.